### PR TITLE
Tokenizer config fix for dynamic mode (#144)

### PR DESCRIPTION
**Pull request by pramodkumar-habanalabs**
Issue:
For the Llama2 7B Dynamic mode, with padding="max_length", the tokenizer produces 'input_ids' without padding and having actual tokens only in the list. This is expected and works fine.

For the Llama3.1 8B Dynamic mode, with padding="max_length", the tokenizer produces 'input_ids' with padding and having 128K tokens (which is the max model input length. This is not expected.

Fix: The tokenizer configuration has been changed to address this issue.

### CODE REVIEW

**Comment by regisss** [comment]
@pramodkumar-habanalabs Can you share a code snippet that enables me to reproduce this issue with Llama 3.1 please?

**Comment by libinta** [comment]
@regisss You can try with below 
PT_HPU_LAZY_MODE=0 python3 /root/repos/optimum-habana-fork/examples/text-generation/run_generation.py --model_name_or_path /mnt/weka/data/pytorch/llama3.1/Meta-Llama-3.1-8B-Instruct/  --attn_softmax_bf16  --use_kv_cache --max_new_tokens 128 --bf16 --batch_size 8 --trim_logits --max_input_tokens -1  --warmup 2  --torch_compile --dataset_name tatsu-lab/alpaca --run_partial_dataset --n_iterations 10

**Review by regisss — APPROVED** [review]
LGTM
